language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/InterfaceClass.scala

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.codegen.backend.java.inner
+
+import com.daml.ledger.javaapi
+import com.daml.lf.codegen.backend.java.JavaEscaper
+import com.daml.lf.data.Ref.{PackageId, QualifiedName}
+import com.daml.lf.iface.{DefInterface}
+import com.squareup.javapoet._
+import com.typesafe.scalalogging.StrictLogging
+
+import javax.lang.model.element.Modifier
+
+object InterfaceClass extends StrictLogging {
+
+  def classNameForInterface(qualifiedName: QualifiedName) =
+    ClassName.bestGuess {
+      val QualifiedName(module, name) = qualifiedName
+      // consider all but the last name segment to be part of the java package name
+      val packageSegments = module.segments.slowAppend(name.segments).toSeq.dropRight(1)
+      // consider the last name segment to be the java class name
+      val className = name.segments.toSeq.takeRight(1)
+
+      val packageName = packageSegments.map(_.toLowerCase)
+
+      (packageName ++ className)
+        .filter(_.nonEmpty)
+        .map(JavaEscaper.escapeString)
+        .mkString(".")
+    }
+
+  def generate(
+      interfaceName: ClassName,
+      interface: DefInterface.FWT,
+      packagePrefixes: Map[PackageId, String],
+      packageId: PackageId,
+      interfaceId: QualifiedName,
+  ): TypeSpec =
+    TrackLineage.of("interface", interfaceName.simpleName()) {
+      logger.info("Start")
```

Comment:
Should this be info?

Reply:
honestly don't know, I focused on keeping the logic and doing only small refactoring (thus no semantic changes).

Reply:
Ok, yeah it just looked like a testing left-over, changing a debug into info and then leaving it around.. (I'm not sure who reads these logs so maybe this could be changed to debug, maybe @stefanobaghino-da @stephencompall-DA know. (it's super minor of course)